### PR TITLE
Bump gatsby to 2.24.42.

### PR DESCRIPTION
**Pull request by jcastle**
## Description

Bump gatsby from 2.24.41 to current version of 2.24.42 in package.json. Resolves issue #26412.
Update tutorial documentation for creating a new gatsby site with hello-world starter files.

### CODE REVIEW

**Comment by jcastle** [comment]
Looks like gatsby dependencies need updating too. I didn't go into gatsby package.json file thinking that would be an upstream change.

File: docs/tutorial/part-zero/index.md [hunk 1]
```diff
@@ -211,7 +211,7 @@ Now you are ready to use the Gatsby CLI tool to create your first Gatsby site. U
 2. Create a new site from a starter:
 
 ```shell
-gatsby new hello-world https://github.com/gatsbyjs/gatsby-starter-hello-world
+gatsby new hello-world https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world
```
Comment (pvdz) [inline docs/tutorial/part-zero/index.md:214]: Can you elaborate on this change? It seems unnecessary as I just tested the old version (`gatsby new hello-world https://github.com/gatsbyjs/gatsby-starter-hello-world`) and that works fine. If nothing else, please undo this line.

File: starters/gatsby-starter-theme/package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
   },
   "license": "0BSD",
   "dependencies": {
-    "gatsby": "^2.24.41",
+    "gatsby": "^2.24.42",
```
Comment (pvdz) [inline starters/gatsby-starter-theme/package.json:13]: This project has no lock file and the `^` should tell `npm` and `yarn` to install the latest known patch version. In other words, there's no real need to bump the patch version for every time we update the gatsby package. Please undo this change.

Comment (vladar) [reply starters/gatsby-starter-theme/package.json:13]: @pvdz Sadly the project has a lock file and it contains `2.24.41` in it: https://github.com/gatsbyjs/gatsby/blob/master/starters/gatsby-starter-theme/package-lock.json

**Comment by pvdz** [comment]
As pointed out by @vladar, the version bump is fine due to lock. Sorry about that. Can you please undo the change to the url (or explain it)? Thanks!

File: docs/tutorial/part-zero/index.md (hunk 1 above)
Comment (jcastle) [reply docs/tutorial/part-zero/index.md:214]: The reason for the URL update is to point to the current repo. While the old link still works, it points to a repo where I received a message saying it would no longer be maintained and to look at the newest documentation on the new site. I imagine all starter files need to be updated but I was only working with the hello-world one.

Comment (pvdz) [reply docs/tutorial/part-zero/index.md:214]: Great point. Thank you!

**Comment by jcastle** [comment]
Thanks @pvdz for the merge! This will help me and future users of hello world. My work around was an updated forked repo.

File: docs/tutorial/part-zero/index.md (hunk 1 above)
Comment (muescha) [reply docs/tutorial/part-zero/index.md:214]: the https://github.com/gatsbyjs/gatsby-starter-hello-world should be a mirror of https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world - if they are not in sync then this should be checked

Comment (muescha) [reply docs/tutorial/part-zero/index.md:214]: and i got this error:

```shell
$ gatsby new hello-world https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world

 ERROR 

starter https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world doesn't exist



  Error: starter https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world doesn't exist
  
  - init-starter.js:157 copy
    [lib]/[gatsby-cli]/lib/init-starter.js:157:11
  
  - init-starter.js:345 async initStarter
    [lib]/[gatsby-cli]/lib/init-starter.js:345:58
  
  - create-cli.js:400 
    [lib]/[gatsby-cli]/lib/create-cli.js:400:7

```

Comment (muescha) [reply docs/tutorial/part-zero/index.md:214]: is see there are already a PR for this:

- #26504 `In Tutorial Docs Step 0, wrong git clone URL is used`

**Comment by muescha** [comment]
and i got this error:

```shell
$ gatsby new hello-world https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world

 ERROR 

starter https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world doesn't exist



  Error: starter https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world doesn't exist
  
  - init-starter.js:157 copy
    [lib]/[gatsby-cli]/lib/init-starter.js:157:11
  
  - init-starter.js:345 async initStarter
    [lib]/[gatsby-cli]/lib/init-starter.js:345:58
  
  - create-cli.js:400 
    [lib]/[gatsby-cli]/lib/create-cli.js:400:7

```

is see there are already some PRs to fix this:

- #26504 `In Tutorial Docs Step 0, wrong git clone URL is used`
- #26510 `Update index.md : Change hello world starter link`

File: docs/tutorial/part-zero/index.md (hunk 1 above)
Comment (pieh) [reply docs/tutorial/part-zero/index.md:214]: Touching on this - the automation that we used to sync read-only repositories was broken since Aug 12th. I fixed the sync automation yesterday, so this change should be reverted - and the https://github.com/gatsbyjs/gatsby/pull/26528 seems to be doing that
